crates/uv-build-frontend/src/error.rs

```diff
@@ -68,19 +69,47 @@ pub enum Error {
     #[error("Editable installs with setup.py legacy builds are unsupported, please specify a build backend in pyproject.toml")]
     EditableSetupPy,
     #[error("Failed to resolve requirements from {0}")]
-    RequirementsResolve(&'static str, #[source] anyhow::Error),
+    RequirementsResolve(&'static str, #[source] AnyErrorBuild),
     #[error("Failed to install requirements from {0}")]
-    RequirementsInstall(&'static str, #[source] anyhow::Error),
+    RequirementsInstall(&'static str, #[source] AnyErrorBuild),
```

Comment:
Same as below, but this might be dropping any chain that might have been on `anyhow::Error`? Not 100% sure

Reply:
`AnyErrorBuild` preserves the chain, it passes `.source()` calls through to its wrapped type.

Reply:
I mean its `Display` impl. The `Display` impl for a `Box<dyn Error>` doesn't emit the chain. But a `anyhow::Error` will.

Reply:
afaik all the error types only show their inner message, but not the chain, over which you have to iterate manually. When testing,

```rust
use thiserror::Error;

#[derive(Error, Debug)]
#[error("Wrapper Err")]
struct WrappedErr;

fn main() {
    let err: anyhow::Error = WrappedErr.into();
    println!("{}", err);
    println!("{}", err.context("2"));
}
```

shows

```
Wrapper Err
2
```

and we don't seem to be losing and context in our snapshot tests.

Reply:
anyhow does show the chain, but only in alternate mode:

```rust
use thiserror::Error;

#[derive(Error, Debug)]
#[error("Inner Err")]
struct InnerErr;

#[derive(Error, Debug)]
#[error("Wrapper Err")]
struct WrappedErr(#[source] InnerErr);

fn main() {
    let err = WrappedErr(InnerErr);
    println!("{:#}", err);
    let err: anyhow::Error = WrappedErr(InnerErr).into();
    println!("{:#}", err);
    println!("{:#}", err.context("2"));
}
```

shows

```
Wrapper Err
Wrapper Err: Inner Err
2: Wrapper Err: Inner Err
```

Reply:
Ah okay, so `anyhow::Error`'s `Display` impl is the same as `Box<dyn Error>`. Interesting. All righty.

And yeah, I realize the snapshot tests indicate nothing untoward is happening here. I don't have a full grasp of the surrounding code and I wasn't sure how easy it would be to break something in the future. But I think you've demonstrated my concern is unfounded. Thank you. :)